doc/event-alarm-framework/event-alarm-framework.md

```diff
@@ -0,0 +1,1100 @@
+ 1) "id"
+ 2) "29"
+ 3) "type-id"
+ 4) "TEMPERATURE_EXCEEDED"
+ 5) "text"
+ 6) "temperatureCrossedThreshold: Current temperature for sensor/2 is 76 degrees"
+ 7) "resource"
+ 8) "sensor/2"
+ 9) "time-created"
+10) "2021-02-10.04:51:53.454450"
+11) "is-acknowledged"
+12) "false"
+13) "severity"
+14) "critical"
+127.0.0.1:6379[6]>
+```
+
+Schema for ALARM_STATS table is as below. When an alarm of particular severity is cleared,
+the corresponding severity counter is decremented. 
+```
+ALARM_STATS Table:
+==============================
+
+Key                       : id
+
+id                        : key {state}
+alarms                    : Number of active alarms {uint64}
+critical                  : Number of alarms of severity 'critical' {uint64}
+major                     : Number of alarms of severity 'major' {uint64}
+minor                     : Number of alarms of severity 'minor' {uint64}
+warning                   : Number of alarms of severity 'warning' {uint64}
+
+127.0.0.1:6379[6]> hgetall "ALARM_STATS|state"
+ 1) "alarms"
+ 2) "1"
+ 3) "critical"
+ 4) "0"
+ 5) "major"
+ 6) "1"
+ 7) "minor"
+ 8) "0"
+ 9) "warning"
+10) "0"
+
+```
+
+System Health Table
+```
+SYSTEM_HEALTH Table:
+==============================
+key           : “SYSTEM_STATE”
+state         : enum to indicate the health state of the system (NORMAL, WARNING, ERROR) 
+
+```
+### 3.1.8 Pull Model
+All NBIs - CLI, REST and gNMI - can pull contents of current alarm table and event history table.
+The following filters are supported:
+- ALL ( pulls all alarms)
+- Severity.
+- Recent alarms (eg., last 5 minutes, one hour, one day).
+- Records between two timestamps, one timestamp and end, and   beginning and a timestamp.
+- All records between two Sequence Numbers (incl end and begin)
+
+### 3.1.9 Supporting third party containers
+To support third party components ( e.g. FRR, teamd, DHCP Relay, LLDPd, ntpd etc ) which can not be modified to raise events, the following options are considered 
+and are being evaluated.
+1.  Patch the components
+    Create a patch for these components by adding libeventnotify library and invoke the API. This however, requires these patches need to be maintained in the code forever.
+
+2.  Listen to syslog messages
+    As many of these components raises syslog messages on an important event, a listener can be implemented to read incoming syslog messages and raise 
+    events based on the message.
+    This however is heavy on performance due to the fact that listener has to parse each syslog message. Also listener need to maintain a map of messages to 
+    event-id and need to be aware of resource and other specific details. It need to be aware of nuances of alarm raising/clearing if the component follows
+    any specific logic. 
+
+Approach 1 is preferred.
+
+## 3.2 DB Changes
+### 3.2.1 CONFIG DB
+### 3.2.2 APP DB
+Table EVENTPUBSUB is used for applications to write events and for eventd to access and process them.
```

Comment:
If i am not mistaken you mentioned this is stored after the system restart as of that I don't understand how you store it in APP_DB. Can you please elaborate?

Reply:
The pusbub table which is used by applications to publish events and eventd to receive them  is "housed" in app-db. 
HOwever, as per the review comments, I am going to migrate this to a new DB specifically for event framework. I will update the HLD.

Reply:
All the tables used in event framework will be housed in new DB called EVENT_DB

Reply:
The community has comment that the tables need to have separate DB for event/alarm tables. 
Does it mean I need to have a new redis instance or a new DB in one of the existing instances?